### PR TITLE
Add new query to enumerate users, mutation to change ban status

### PR DESCRIPTION
**Pull request by forabi**

### CODE REVIEW

**Comment by forabi** [comment]
This PR has become quite large, I'm going to merge it and work on quote management in a separate PR. I've also relaxed the "find-breaking-changes" script to just print warning instead of failing the build. I think that's okay for the time being.

File: src/database/entities/Photo.ts [hunk 1]
```diff
@@ -38,7 +39,11 @@ export class Photo extends BaseEntity {
   @Column({ type: 'varchar', nullable: true })
   credits: string | null;
 
-  @Column({ type: 'tinyint', nullable: true })
+  @Column({
+    type: 'tinyint',
+    nullable: true,
+    transformer: transformTinyIntOrNull,
```
Comment (wholesomedev) [inline src/database/entities/Photo.ts]: What's the benefit of using `transformTinyIntOrNull`?

File: src/resolvers/mutations/changeUserIsBannedStatus.ts [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+      user.isBanned = newValue;
+
+      await users.save(user);
+
+      return {
+        result: makeSuccessResult(),
+      };
+    },
+  },
+};
```
Comment (wholesomedev) [inline src/resolvers/mutations/changeUserIsBannedStatus.ts]: Nice attention to edge cases!

File: src/helpers/createConnectionResolverFromEntity.ts [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+    return {
+      edges,
+      pageInfo: {
+        hasNextPage: skip + nodes.length < all,
+        endCursor: edges.length > 0 ? edges[edges.length - 1].cursor : null,
+      },
+    };
+  };
+}
```
Comment (wholesomedev) [inline src/helpers/createConnectionResolverFromEntity.ts]: Good idea to abstract this code now that it's needed in multiple places 👍

File: src/schema.graphql [hunk 4]
```diff
@@ -211,6 +214,21 @@ type NotablePersonConnection {
...
+  node: User!
+}
+
+type UserConnection {
+  edges: [UserEdge!]!
+  pageInfo: PageInfo!
+}
+
+input UsersQueryWhereInput {
+  # If specified, users will be filtered based on whether ther are banned or not
```
Comment (wholesomedev) [inline src/schema.graphql]: Typo

File: src/database/entities/Photo.ts (hunk 1 above)
Comment (forabi) [reply src/database/entities/Photo.ts]: I merged this before I documented this. I've included a comment to explain why we need this in #106.

File: src/schema.graphql (hunk 4 above)
Comment (forabi) [reply src/schema.graphql]: Fixed in #102.

**Pull request merged**